src/transformers/models/jukebox/modeling_jukebox.py

```diff
@@ -0,0 +1,2694 @@
+        tokens = []
+        iterator = tqdm(zip(music_tokens_list, music_tokens_conds_list, metadata_list), leave=False)
+        for music_tokens_i, music_tokens_conds_i, metadata_i in iterator:
+            iterator.set_description(f"Sampling windows of {sample_tokens}")
+            tokens_i = prior.sample(
+                n_samples=music_tokens_i.shape[0],
+                music_tokens=music_tokens_i,
+                music_tokens_conds=music_tokens_conds_i,
+                metadata=metadata_i,
+                **sampling_kwargs,
+            )
+            tokens.append(tokens_i)
+        sampled_tokens = torch.cat(tokens, dim=0)
+
+        # Update music_tokens with new sample
+        music_tokens_new = sampled_tokens[:, -new_tokens:]
+        music_tokens[level] = torch.cat([music_tokens[level], music_tokens_new], dim=1)
+        return music_tokens
+
+    # Sample total_length tokens at level=level with hop_length=hop_length
+    def sample_level(self, music_tokens, labels, offset, sampling_kwargs, level, total_length, hop_length):
+        if total_length >= self.priors[level].n_ctx:
+            iterator = tqdm(get_starts(total_length, self.priors[level].n_ctx, hop_length), leave=False)
+            for start in get_starts(total_length, self.priors[level].n_ctx, hop_length):
+                iterator.set_description(
+                    f"[prior level {level}] Sampling {self.priors[level].n_ctx}/{total_length} tokens", refresh=True
+                )
+                music_tokens = self.sample_single_window(music_tokens, labels, offset, sampling_kwargs, level, start)
+
+        else:
+            music_tokens = self.sample_partial_window(
+                music_tokens, labels, offset, sampling_kwargs, level, total_length
+            )
+        return music_tokens
+
+    @torch.no_grad()
+    def _sample(
+        self,
+        music_tokens,
+        labels,
+        sample_levels,
+        metas=None,
+        chunk_size=32,
+        sampling_temperature=0.98,
+        lower_batch_size=16,
+        max_batch_size=16,
+        sample_length_in_seconds=24,
+        compute_alignments=False,
+        sample_tokens=None,
+        offset=0,
+        save_results=True,
+        sample_length=None,
+    ) -> List[torch.LongTensor]:
+        """
+        Core sampling function used to generate music tokens. Iterates over the provided list of levels, while saving
+        the generated raw audio at each step.
+
+        Args:
+           music_tokens (`List[torch.LongTensor`] of length `self.levels` ) :
+                A sequence of music tokens which will be used as context to continue the sampling process. Should have
+                `self.levels` tensors, each corresponding to the generation at a certain level.
+            labels (`List[Torch.LongTensor]` of lenght `n_sample`, and shape `(self.levels, 4 +
+            self.config.max_nb_genre + lyric_sequence_lenght)` :
```

Comment:
```suggestion
            labels (`List[Torch.LongTensor]` of lenght `n_sample`, and shape `(self.levels, 4 + self.config.max_nb_genre + lyric_sequence_lenght)` :
```
Needs to be on one line.